litex/build/microsemi/libero_soc.py

```diff
@@ -208,19 +211,23 @@ def build_timing_constraints(self, vns):
         return (self._build_name + ".sdc", "SDC")
 
     # Script ---------------------------------------------------------------------------------------
-
+   
     def build_script(self):
         if sys.platform in ("win32", "cygwin"):
             script_ext = ".bat"
             script_contents = "@echo off\nREM Autogenerated by LiteX / git: " + tools.get_litex_git_revision() + "\n\n"
             copy_stmt = "copy"
             fail_stmt = " || exit /b"
         else:
-            script_ext = ".sh"
+            #script_ext = ".sh"
             script_contents = "# Autogenerated by LiteX / git: " + tools.get_litex_git_revision() + "\n"
+            if os.getenv("LITEX_ENV_LIBERO", False):
+            	script_contents += "source " + os.path.join(os.getenv("LITEX_ENV_LIBERO"))
+            script_ext = ".sh"
```

Comment:
why moving `script_ext` ?
you use a `source` for a script, but unlike vivado or efinity there is nothing to source to update current shell's env.
Also, for vivado, *LITEX_ENV_XXX* is appended by a script name, here you assume this variable contains directly the script name.

Reply:
Hi @trabucayre,

I thought I had an issue with script_ext so I moved it. Reverted back to where is was before and check it and it works ok. I'll change that now and check it it in.  
On the LITEX_ENV_XXX issue, what would your recommendation be?

Reply:
Unlike vivado, libero didn't provides a script to source. This variable isn't required, so you have to consider libero is in *PATH* (and `which` is used to check that).

Reply:
hi @trabucayre ,
Apologies I was away on vacation for a couple of weeks and I am only getting back to this now. I can remove that part of the code and make sure to have a comment saying that Libero must be in PATH. I can try get this change made over the next couple of days if that works and I am on the right track with what I am saying.

Reply:
I have removed the Libero environment variable from the pull request.